etcd.go

```diff
@@ -0,0 +1,127 @@
+package dsync
+
+import (
+	"github.com/coreos/etcd/client"
+	"github.com/pborman/uuid"
+	"golang.org/x/net/context"
+	"sync"
+	"time"
+)
+
+type etcdMutex struct {
+	backoff BackoffFunc
+	ctx     context.Context
+	etcd    client.KeysAPI
+	key     string
+	locked  bool
+	m       sync.Mutex
+	refresh time.Duration
+	stop    chan struct{}
+	ttl     time.Duration
+	uuid    string
+	wg      sync.WaitGroup
+}
+
+func (e *etcdMutex) Lock() {
+	var i int64
+	e.m.Lock()
```

Comment:
it might be good to use a defer here for the UnLock(). The UnLock is so far away I am worried it might be missed. That being said you have a security issue. This lock is in place while you wait in the select/case. Which means everything is frozen at it related to this lock. This is raising flags.